### DISCUSSION TITLE
[firebase_dynamic_links] allow custom paths for short URLs

### DISCUSSION THREAD

**Discussion by gisinator** [firebase/flutterfire#5573] [Feature request]
**Is your feature request related to a problem? Please describe.**
I can create short dynamic links with my own custom domain no problem. They look like "https://app.myapp.com/2dBY" for example and link a user within my app. Problem: "2dBY" doesnt look cool. Within firebase console, custom paths are possible, so my link could look like "https://app.myapp.com/Chris123" for example. In flutter, I did not find a way to do this.

**Describe the solution you'd like**
It would be cool if "DynamicLinkParameters class" could be extended with a new parameter called "urlSuffix" for example. This suffix would be used for short dynamic links, so that even within flutter I could generate userfriendly short links like "https://app.myapp.com/Chris123"

**Additional context**
Guess (and hope) that no more info is required :)

**Comment by edisonpagtakhan** [comment]
I think it's a really great feature to add. However, I realized that there are a lot of things to consider. When building a new dynamic link using the api, it's trying creating a new one. What if the custom path already exists? And how do we make sure that the new and existing paths have the same link?

**Comment by simbakowo** [comment]
If the program is using usernames, which are unique(the app has to deal with this), and the same user generates a new dynamic link again, why not just overwrite the old link . The custon path feature would  be very useful it'll allow user to generate better looking links to get other users to signup eg. instagram/someusername

**Comment by Vaytrex** [comment]
For anyone landing here after the Aug 2025 shutdown: there's no drop-in. You need a URL that branches on User-Agent (App Store / intent:// with browser_fallback_url / website) and skips redirecting crawlers so previews survive. Full write-up with the intent format: https://linkbah.app/firebase-dynamic-links-alternative
